modules/brushingandlinking/include/modules/brushingandlinking/brushingandlinkingmanager.h

```diff
@@ -30,73 +30,216 @@
+     */
+    BrushingModifications modifiedActions() const;
+    /**
+     * return whether there was a filter action since the last network evaluation
+     *
+     * @return modifiedActions() & BrushingModification::Filtered
+     */
+    bool modifiedFiltering() const;
+    /**
+     * return whether there was a select action since the last network evaluation
+     *
+     * @return modifiedActions() & BrushingModification::Selected
+     */
+    bool modifiedSelection() const;
+    /**
+     * return whether there was a highlight action since the last network evaluation
+     *
+     * @return modifiedActions() & BrushingModification::Highlighted
+     */
+    bool modifiedHighlight() const;
+
+    /**
+     * check whether the manager has an index set for \p target and \p action
+     *
+     * @param action   type of brushing action
+     * @param target   target of the action
+     * @return true if indices exist
+     */
+    bool hasIndices(BrushingAction action, BrushingTarget target = BrushingTarget::Row) const;
+
+    /**
+     * access indices for the given combination of \p action and \p target
+     *
+     * @param action    type of brushing action
+     * @param target    target of the action
+     * @return bitset corresponding to \p action and \p target
+     * @throw Exception if \p target does not exist for \p action
+     *
+     * \see hasIndices
+     */
+    const BitSet& getIndices(BrushingAction action,
+                             BrushingTarget target = BrushingTarget::Row) const;
+
+    /**
+     * clear the selection for \p action and \p target. \p action must be different from
+     * BrushingAction::Filter. Does nothing if \p target does not exist.
+     *
+     * @throw Exception if action is equal to BrushingAction::Filter since this may lead to an
+     * inconsistent state
+     */
+    void clearIndices(BrushingAction action, BrushingTarget target);
 
-    bool isFiltered(size_t idx) const;
-    bool isSelected(size_t idx) const;
+    /**
+     * check whether the selection for \p action and \p target contains index \p idx
+     */
+    bool contains(uint32_t idx, BrushingAction action,
+                  BrushingTarget target = BrushingTarget::Row) const;
+
+    std::vector<std::pair<BrushingAction, BrushingTarget>> getTargets() const;
+    std::vector<BrushingTarget> getTargets(BrushingAction action) const;
+
+    /**
+     * update the internal state with respect to port connections. This function should be called
+     * whenever a connection is removed from a brushing and linking outport. This will only effect
+     * filter actions and has no effect if the manager is owned by an inport.
+     *
+     * \see action
+     */
+    void updatePortConnections();
+
+    //! convenience function for action(BrushingAction::Filter, target, idx)
+    void filter(std::string_view src, const BitSet& idx,
+                BrushingTarget target = BrushingTarget::Row);
```

Comment:
```suggestion
    void filter(const BitSet& idx, BrushingTarget target, std::string_view src);
```
To be consistent with brush